website/cue/reference/components/sources/kubernetes_logs.cue

```diff
@@ -42,6 +42,11 @@ components: sources: kubernetes_logs: {
 			"""
 				[Kubernetes](\(urls.kubernetes)) version `\(services.kubernetes.versions)` is required.
 				""",
+			"""
+				This source reads log files from the host, *not* the Kubernetes API. If Vector is running within Kubernetes,
+				the directory `/var/log/pods` must be mounted into the Vector container using a `hostPath` volume. More information
+				about `hostPath` volumes can found [here](\(urls.kubernetes_host_path)).
```

Comment:
```suggestion
				This source requires read access to the `/var/log/pods` directory.
```

I think this is more accurate to the requirement, and more in-line with the [file source](https://vector.dev/docs/reference/configuration/sources/file/#requirements).

Your point did make me think that the [api access control](https://vector.dev/docs/reference/configuration/sources/kubernetes_logs/#kubernetes-api-access-control) section is similarly commentary on requirements, same as needing access to the container runtimes logging directory. 🤔

Reply:
I personally think that the volume config information helps, since it's going to be relevant in almost all cases 😸 

One other note is that most pods will have read access to `/var/log/pods` by default, although not all will have the volume mount configured with the host, so nothing will be there. I just think that a lot of info here is good to avoid confusion. 

I know we're adding a lot of information to the top of this page, but this source is pretty unique in that the environment Vector runs on needs specific configuration. Someone in the Discord and I both wrongly assumed that this source worked via the Kubernetes API, without a dependency on the host, since there was no mention of volume mounts on the page. 

Let me know what you think. Ultimately it's your call since it's your source 😄

Reply:
> Let me know what you think. Ultimately it's your call since it's your source 😄

Don't put that evil on me!

I definitely agree the documentation wasn't great, in fact the current site (I've updated as part of the `configurable` PRs) claims:

> Collects all log data for Kubernetes Nodes...

Which is super inaccurate...

> One other note is that most pods will have read access to `/var/log/pods` by default...

Haha, I mean yes that's technically accurate 🤔. So, I'm in two minds about this. On the one hand 95% of the usage of this source will be happening while deployed _in_ Kubernetes, so it makes sense to cater the documentation to that (regarding `hostpath`) - however it's not _strictly_ required as you could run it on the Kubernetes notes on the actual host and it's normal Linux permissions to those files.

A blend between the two could look something like

```markdown
This source requires read access to the `/var/log/pods` directory. When run in a
Kubernetes cluster this can be provided with a [hostPath](https://kubernetes.io/link) volume.
```

Re: directly running directly through the API, I'm definitely biased having used/owned the source so much but I felt like the docs made it fairly clear that the _enrichment_ portion is driven by the API not the collection - it's less explicit that the collection is file based. So that's another place we could make improvements.

Reply:
I went with your suggestion above. Thanks!